### PR TITLE
ddl, test : fix CI with adding new tk to execute additional sql.

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve? 
Fix [CI problem](https://internal.pingcap.net/idc-jenkins/blue/rest/organizations/jenkins/pipelines/tidb_ghpr_unit_test/runs/15267/nodes/75/steps/278/log/?start=0) caused by #13119

### What is changed and how it works?
**Background**:
 In the ddl state change callback, we conduct additional sql to show create table, in order to make sure the schema state is the same with what we want it to be.  

**Root Cause** : 
This will cause `resetContextOfStmt` of the same `tk`, consequently, causing data race.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to cherry-pick to the release branch

Release note

 - fix CI with adding new tk to execute additional sql in `TestShowCreateTable`.

### CODE REVIEW

File: ddl/db_change_test.go [hunk 1]
```diff
@@ -114,9 +118,9 @@ func (s *testStateChangeSuite) TestShowCreateTable(c *C) {
 			var result *testkit.Result
 			tbl2 := testGetTableByName(c, tk.Se, "test", "t2")
```
Comment (zimulala) [inline ddl/db_change_test.go]: Do we need to replace `tk.Se`with `tkInternal.Se`? Otherwise, it also has DATA RACE.

Comment (AilinKid) [reply ddl/db_change_test.go]: Nice idea, `testGetTableByName` won't exec SQL, but get the domain and check the schema.
Both `tk` and `tkInternal` is ok, maybe tkInternal is more elegant, Addressed.

File: ddl/db_change_test.go [hunk 2]
```diff
@@ -111,14 +115,27 @@ func (s *testStateChangeSuite) TestShowCreateTable(c *C) {
...
-			var result *testkit.Result
-			tbl2 := testGetTableByName(c, tk.Se, "test", "t2")
+			var result sqlexec.RecordSet
+			var err error
+			tbl2 := testGetTableByName(c, tkInternal.Se, "test", "t2")
 			if job.TableID == tbl2.Meta().ID {
-				result = tk.MustQuery("show create table t2")
+				// Try to do not use mustQuery in hook func, cause assert fail in mustQuery will cause ddl job hung.
+				result, err = tkInternal.Exec("show create table t2")
+				if err != nil {
```
Comment (zimulala) [inline ddl/db_change_test.go]: I think it's not useful. We need to turn it to `checkErr`.

Comment (AilinKid) [reply ddl/db_change_test.go]: good catch!

File: ddl/db_change_test.go [hunk 3]
```diff
@@ -83,6 +83,10 @@ func (s *testStateChangeSuite) TestShowCreateTable(c *C) {
 	tk.MustExec("use test")
 	tk.MustExec("create table t (id int)")
 	tk.MustExec("create table t2 (a int, b varchar(10)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci")
+	// tkInternal is used to execute additional sql (here show create table) in ddl change callback.
+	// Otherwise resetContextOfStmt in same tk will cause data race sometimes.
```
Comment (bb7133) [inline ddl/db_change_test.go]:
```suggestion
	// Using same `tk` in different goroutines may lead to data race.
```

Comment (AilinKid) [reply ddl/db_change_test.go]: Good catch, Addressed.

**Pull request merged**